tests/nested/manual/core20-initramfs-time-moves-forward/task.yaml

```diff
@@ -0,0 +1,39 @@
+summary: Test that time moves forward when the RTC is broken/unavailable in UC20
+
+systems: [ubuntu-20.04-64]
+
+environment:
+  NESTED_IMAGE_ID: core20-initramfs-time-moves-forward
+  NESTED_BUILD_SNAPD_FROM_CURRENT: true
+  NESTED_USE_CLOUD_INIT: true
+
+  # disable the real time clock and effectively break it so it returns unix
+  # epoch to simulate devices like the rpi which do not have RTC
+  NESTED_PARAM_RTC: "-rtc base=1970-01-01,clock=vm"
```

Comment:
Is the VM also running without network? As otherwise upon first boot to run-mode, the VM will gain networking and will gain NTP over the network and store that into the VMs rtc clock. Hence the time will be good after the first boot.

If you want the machine to keep on running in 1970 through reboots, you need to somehow boot the machine with systemd.mask=systemd-timesyncd.service or disable networking to the public internet. I.e. only keep the VM on a private network connection (such that you can ssh in) but such that it is not NAT masquaraded to see public NTP.

Reply:
No, we are providing the VM network access since we need to SSH into it, but yeah your suggestion about having a private network connection such that the VM can't reach the full internet but we can still SSH to it is a good one. Do you have an example of how to do that using bare qemu command line (i.e. not with virsh)?

Reply:
@anonymouse64 i don't know how you launch your VMs, by default qemu will create user networking with slirp

aka `-net user`

To restrict public network access one can do `restrict=on` option.
To forward ssh port `hostfwd=tcp::2222-:22` option to access ssh of the guest via port 2222 on the host.

So `-net user,restrict=on,hostfwd=tcp::2222-:22` should keep the VM isolated (no access to internet), yet one can still ssh into it via localhost on port 2222.

For more details see https://qemu-project.gitlab.io/qemu/system/invocation.html#hxtool-5

How is the networking setup currently in the VM? you must be doing portforward already I would have thought.

Reply:
ah thanks, I wasn't aware of the resrict=on option, but yes we are doing slirp via `-net user`. I'll give that a shot and see if that gets the spread test working the way I wanted it to

Reply:
@xnox Even with `restrict=on` option set so that the VM doesn't have internet access the VM still is able to move the time forward in the initrd by talking to _some_ RTC. This is the full QEMU command line I used:

```
/usr/bin/qemu-system-x86_64 \
    -smp 2 \
    -cpu host \
    -m 4096 \
    -d cpu_reset \
    -D /tmp/work-dir/logs/qemu.log \
    -rtc base=1970-01-01,clock=vm \
    -machine ubuntu,accel=kvm \
    -nographic \
    -net nic,model=virtio \
    -net user,restrict=on,hostfwd=tcp::8022-:22 \
    -bios /usr/share/ovmf/OVMF.fd \
    -object rng-random,id=rng0,filename=/dev/urandom \
    -device virtio-rng-pci,rng=rng0 \
    -drive file=/tmp/work-dir/images/ubuntu-core-current.img,cache=none,format=raw,id=disk1,if=none \
    -device virtio-blk-pci,drive=disk1,bootindex=1 \
    -chardev socket,telnet,host=localhost,server,port=7777,nowait,id=char0,logfile=/tmp/work-dir/logs/serial.log,logappend=on \
    -serial chardev:char0 \
    -monitor tcp:127.0.0.1:8888,server,nowait \
    -usb
```

Just as an FYI to other reviewers, we should land this PR even if we can't improve the nested spread test to directly test what I wanted to with the timestamps before/after running snap-bootstrap in the initrd.